programs/drift/src/controller/orders.rs

```diff
@@ -1899,12 +1907,14 @@ pub fn fulfill_perp_order_with_amm(
         get_taker_and_maker_for_order_record(user_key, &user.orders[order_index]);
 
     let fill_record_id = get_then_update_id!(market, next_fill_record_id);
-    let order_action_explanation =
-        if override_base_asset_amount.is_some() && override_fill_price.is_some() {
-            OrderActionExplanation::OrderFilledWithAMMJit
-        } else {
-            OrderActionExplanation::OrderFilledWithAMM
-        };
+    let order_action_explanation = match (override_base_asset_amount, override_fill_price) {
+        (Some(_), Some(_)) => match liquidity_split {
```

Comment:
nit: could add `get_order_action_explanation` to AMMLiquiditySplit to make this read a lil simpler